handler/delete.go

```diff
@@ -37,7 +38,12 @@ func Delete(ms storage.MetricStore, logger log.Logger) func(http.ResponseWriter,
 	instrumentedHandler := promhttp.InstrumentHandlerCounter(
 		httpCnt.MustCurryWith(prometheus.Labels{"handler": "delete"}),
 		http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
-			job := ps.ByName("job")
+			job, err := maybeDecodeBase64(ps.ByName("job"))
+			if err != nil {
+				http.Error(w, fmt.Sprintf("invalid base64 encoding in job name %q: %v", ps.ByName("job"), err), http.StatusBadRequest)
+				level.Debug(logger).Log("msg", "invalid base64 encoding in job name", "job", ps.ByName("job"), "err", err.Error())
```

Comment:
Should this be higher than debug?

Reply:
The idea here is this is an error of the pushing client (not an error in the PGW), and the error message is already sent back in the HTTP response to the pushing client. We are not really doing query logging in the PGW, and this would be more query logging.

However, I do not feel strongly about this distinction. However, if we decided to log this at a higher level, we should change it everywhere else where similar things are done (e.g. two more instances here in the very same handler function and many more in the push handler). We should also log more then (like IP number of the pushing client and such). My concern then would be that we are dealing with a different issue then the one this PR is about, and we should do that in a separate PR.

Reply:
As a data point: When we introduced those log lines, the decision to do it at debug level was deliberate. We can, of course, revisit that decision. Just saying that it is not an accident.